### PR TITLE
Fix dependencies and tests

### PR DESCRIPTION
**Pull request by aszenz**
Follow up of https://github.com/thecodingmachine/schema-analyzer/commit/0ccb1a2a5d1b4434ea613178778d6ecb7be53cdf

To support dbal v3 fully i bumped up the dependencies and php versions

Also added github ci and removed travis

### CODE REVIEW

**Comment by aszenz** [comment]
@nguyenk This pr should make releasing v2 possible

**Comment by aszenz** [comment]
@moufmouf This one is similar to other prs:
- https://github.com/thecodingmachine/dbal-fluid-schema-builder/pull/12
- https://github.com/thecodingmachine/tdbm-fluid-schema-builder/pull/22

File: tests/SchemaAnalyzerTest.php [hunk 1]
```diff
@@ -550,15 +549,12 @@ public function testChainedJunctionTables()
...
-    /**
-     * @expectedException \Mouf\Database\SchemaAnalyzer\SchemaAnalyzerTableNotFoundException
-     * @expectedExceptionMessage Could not find table 'rights'. Did you mean 'right'?
-     */
     public function testWringTableName()
     {
         $schemaManager = $this->getCompleteSchemaManager();
 
         $schemaAnalyzer = new SchemaAnalyzer($schemaManager);
+        $this->expectException(SchemaAnalyzerTableNotFoundException::class, "Could not find table 'rights'. Did you mean 'right'?");
```
Comment (homersimpsons) [inline tests/SchemaAnalyzerTest.php]: I may be wrong, but `"Could not find table 'rights'. Did you mean 'right'?"` should be in an `expectExceptionMessage`, no ?

```suggestion
        $this->expectException(SchemaAnalyzerTableNotFoundException::class);
        $this->expectExceptionMessage("Could not find table 'rights'. Did you mean 'right'?");
```

**Review by homersimpsons — CHANGES_REQUESTED** [review]
Thank you for your work on this ! Just a small concern about a test and then it is ready to merge.

**Comment by aszenz** [comment]
> Thank you for your work on this ! Just a small concern about a test and then it is ready to merge.

Applied the suggestion, the assertion message is different than expectedException message indeed

**Comment by homersimpsons** [comment]
Thank you for your contribution !

I saw you had some open Pull Request targeting your own forks. Maybe those are not ready yet, or the target is not correct:

- https://github.com/aszenz/magic-query/pull/1
- https://github.com/aszenz/schema-version-control/pull/1

**Comment by aszenz** [comment]
> Thank you for your contribution !
> 
> I saw you had some open Pull Request targeting your own forks. Maybe those are not ready yet, or the target is not correct:
> 
>     * [Use v2 of schema-analyzer aszenz/magic-query#1](https://github.com/aszenz/magic-query/pull/1)
> 
>     * [Update dbal to v3 aszenz/schema-version-control#1](https://github.com/aszenz/schema-version-control/pull/1)

Those are for testing the ci pipeline, i'll create more pr's in original repos soon

**Comment by aszenz** [comment]
Can u do a release of v2 as well, or something else needs merging

**Comment by homersimpsons** [comment]
> Can u do a release of v2 as well, or something else needs merging

I did a pre-release:
- https://github.com/thecodingmachine/schema-analyzer/releases/tag/v2.0.0-beta.2
- https://packagist.org/packages/mouf/schema-analyzer#v2.0.0-beta.2

If it is stable, I can do a real release. I do not think we have to merge anything else.

**Pull request merged**